src/info/pending.rs

```diff
@@ -12,36 +11,31 @@ pub struct PendingInfo {
 
 impl PendingInfo {
     pub fn new(repo: &Repository) -> Result<Self> {
-        let git_dir = repo.git_dir().to_owned();
-        let repo = git2::Repository::open(git_dir)?;
-        let pending_changes = get_pending_changes(&repo)?;
+        let pending_changes = get_pending_changes(repo)?;
         Ok(Self { pending_changes })
     }
 }
 
-fn get_pending_changes(repo: &git2::Repository) -> Result<String> {
-    let statuses = repo.statuses(Some(
-        StatusOptions::default()
-            .show(StatusShow::Workdir)
-            .update_index(true)
-            .include_untracked(true)
-            .renames_head_to_index(true)
-            .recurse_untracked_dirs(true),
-    ))?;
+fn get_pending_changes(repo: &Repository) -> Result<String> {
+    let statuses = repo
+        .status(gix::progress::Discard)?
+        .dirwalk_options(|options| options.emit_untracked(gix::dir::walk::EmissionMode::Matching))
+        .into_index_worktree_iter(Vec::new())?;
 
-    let (added, deleted, modified) =
-        statuses
-            .iter()
-            .fold((0, 0, 0), |(added, deleted, modified), e| {
-                let s: Status = e.status();
-                if s.is_index_new() || s.is_wt_new() {
-                    (added + 1, deleted, modified)
-                } else if s.is_index_deleted() || s.is_wt_deleted() {
-                    (added, deleted + 1, modified)
-                } else {
-                    (added, deleted, modified + 1)
-                }
-            });
+    let (added, deleted, modified) = statuses
+        .take_while(Result::is_ok)
+        .filter_map(Result::ok)
+        .filter_map(|item| item.summary())
+        .fold((0, 0, 0), |(added, deleted, modified), status| {
+            use gix::status::index_worktree::iter::Summary;
+            match status {
+                Summary::Removed => (added, deleted + 1, modified),
+                Summary::Added | Summary::Copied => (added + 1, deleted, modified),
+                Summary::Modified | Summary::TypeChange => (added, deleted, modified + 1),
+                Summary::Renamed => (added + 1, deleted + 1, modified),
```

Comment:
By the way, it won't run into `Renamed` and `Copied` as rename tracking isn't activated by default.

Reply:
Can you expand on that, what are the implications ?

Reply:
This would mean the code could call `unreachable!()` in case of `Copied` and `Renamed` and be right about it. The implementation/settings have the same effect as `git2` previously.

Does that make more sense?

Reply:
Are you refering to `git config --get diff.renames` ? Whether it's enable doesn't change the output for onefetch, right? a rename is always shown as an addition and an deletion.

Reply:
That's a great point to bring up! I decided not to let it affect renames between index and worktree as Git itself doesn't even expose that functionality to the user (only `git2` seems to have it). Everything that is affected by configuration, really only `emit_untracked` at this time, is overridden though.